src/main/java/com/dabsquared/gitlabjenkins/webhook/build/PushBuildAction.java

```diff
@@ -102,7 +111,64 @@ public void run() {
                     }
                 }
             }
+
+            // TODO configuration can be done via "trigger" like the existing GitlabPushTrigger or similar object
+            // TODO should be replaced with getClass comparison to prevent the dependency to the branch api
+            if (project instanceof MultiBranchProject) {
```

Comment:
This not the right approach. You “just” need to make sure your `SCMRevision` implementation encodes the right things. Look at `github-branch-source` for an example.

Reply:
Thanks for the hints. It is great to see people developing Jenkins here also.

The idea of my use case is to rebuild branches in a multibranch project if there merge target changes.
So I have to track the target branch rev within the SCMRevision implementation to be able to see if a new build is necessary when I trigger the branch indexing?